pkg/securitycontextconstraints/sysctl/mustmatchpatterns.go

```diff
@@ -21,26 +21,88 @@ import (
 	"strings"
 
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/klog/v2"
 	api "k8s.io/kubernetes/pkg/apis/core"
+	utilkernel "k8s.io/kubernetes/pkg/util/kernel"
 )
 
+type sysctl struct {
+	// the name of sysctl
+	name string
+	// the minimum kernel version where the sysctl is available
+	kernel string
+}
+
+var safeSysctls = []sysctl{
+	{
+		name: "kernel.shm_rmid_forced",
+	}, {
+		name: "net.ipv4.ip_local_port_range",
+	}, {
+		name: "net.ipv4.tcp_syncookies",
+	}, {
+		name: "net.ipv4.ping_group_range",
+	}, {
+		name: "net.ipv4.ip_unprivileged_port_start",
+	}, {
+		name:   "net.ipv4.ip_local_reserved_ports",
+		kernel: "3.16",
+	}, {
+		name:   "net.ipv4.tcp_keepalive_time",
+		kernel: "4.5",
+	}, {
+		name:   "net.ipv4.tcp_fin_timeout",
+		kernel: "4.6",
+	}, {
+		name:   "net.ipv4.tcp_keepalive_intvl",
+		kernel: "4.5",
+	}, {
+		name:   "net.ipv4.tcp_keepalive_probes",
+		kernel: "4.5",
+	}, {
+		name:   "net.ipv4.tcp_rmem",
+		kernel: "4.15",
+	}, {
+		name:   "net.ipv4.tcp_wmem",
+		kernel: "4.15",
+	},
+}
+
 // SafeSysctlAllowlist returns the allowlist of safe sysctls and safe sysctl patterns (ending in *).
 //
 // A sysctl is called safe iff
 // - it is namespaced in the container or the pod
 // - it is isolated, i.e. has no influence on any other pod on the same node.
 func SafeSysctlAllowlist() []string {
-	return []string{
-		"kernel.shm_rmid_forced",
-		"net.ipv4.ip_local_port_range",
-		"net.ipv4.tcp_syncookies",
-		"net.ipv4.ping_group_range",
-		"net.ipv4.ip_unprivileged_port_start",
-		"net.ipv4.tcp_keepalive_time",
-		"net.ipv4.tcp_fin_timeout",
-		"net.ipv4.tcp_keepalive_intvl",
-		"net.ipv4.tcp_keepalive_probes",
+	return getSafeSysctlAllowlist(utilkernel.GetVersion)
+}
+
+func getSafeSysctlAllowlist(getVersion func() (*version.Version, error)) []string {
+	kernelVersion, err := getVersion()
+	if err != nil {
+		klog.Error(err, "failed to get kernel version, unable to determine which sysctls are available")
```

Comment:
This would indicates that we don't return a list at all, which is not true.
The message should indicate that we are falling back to safe and working defaults or something like that.

Reply:
At this point we could return the legacy list.